File: src/api/schemas/chunks.py
"""分块管理 Schema"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ChunkListItem(BaseModel):
    """分块列表项"""

    chunk_id: str
    doc_id: str
    chunk_type: str
    page: int
    chunk_index: int
    char_count: int
    full_text: str = Field(description="截断预览，最多 200 字符")
    element_count: int
    group_id: str = ""
    created_at: datetime | None = None


class ChunkListResponse(BaseModel):
    """分块列表响应"""

    total: int
    page: int
    size: int
    items: list[ChunkListItem]


class ChunkDetail(BaseModel):
    """分块详情"""

    chunk_id: str
    doc_id: str
    chunk_type: str
    page: int
    chunk_index: int
    char_count: int
    full_text: str
    elements: list[dict]
    image_urls: list[str] = []
    group_id: str = ""
    created_at: datetime | None = None


class MergeRequest(BaseModel):
    """合并请求"""

    chunk_ids: list[str] = Field(..., min_length=2, description="待合并的 chunk_id 列表，至少 2 个")


class MergeResponse(BaseModel):
    """合并结果"""

    merged_chunk_id: str
    deleted_chunk_ids: list[str]
    char_count: int


class SplitRequest(BaseModel):
    """拆分请求"""

    split_at: int = Field(..., ge=1, description="元素索引：elements[0:split_at] 归 A，elements[split_at:] 归 B")
    link_group: bool = Field(default=False, description="是否将两个子 chunk 关联到同一 group_id")


class SplitChunkInfo(BaseModel):
    """拆分后的分块信息"""

    chunk_id: str
    char_count: int
    element_count: int


class SplitResponse(BaseModel):
    """拆分结果"""

    chunk_a: SplitChunkInfo
    chunk_b: SplitChunkInfo
    deleted_chunk_id: str


class LinkRequest(BaseModel):
    """关联请求"""

    chunk_ids: list[str] = Field(..., min_length=2, description="待关联的 chunk_id 列表，至少 2 个")


class LinkResponse(BaseModel):
    """关联结果"""

    group_id: str
    chunk_ids: list[str]


class UnlinkRequest(BaseModel):
    """取消关联请求"""

    chunk_ids: list[str] = Field(..., min_length=1, description="待取消关联的 chunk_id 列表")


class UnlinkResponse(BaseModel):
    """取消关联结果"""

    unlinked_count: int

File: web/src/api/chunks.ts
import request from './request'

export interface ChunkListItem {
  chunk_id: string
  doc_id: string
  chunk_type: string
  page: number
  chunk_index: number
  char_count: number
  full_text: string
  element_count: number
  image_urls: string[]
  group_id: string
  created_at: string | null
}

export interface ChunkListResponse {
  total: number
  page: number
  size: number
  items: ChunkListItem[]
}

export interface ChunkDetail {
  chunk_id: string
  doc_id: string
  chunk_type: string
  page: number
  chunk_index: number
  char_count: number
  full_text: string
  elements: { type: string; content: string; image_url?: string }[]
  image_urls: string[]
  group_id: string
  created_at: string | null
}

export interface EditChunkResponse {
  chunk_id: string
  full_text: string
  char_count: number
}

export interface MergeResponse {
  merged_chunk_id: string
  deleted_chunk_ids: string[]
  char_count: number
}

export interface SplitResponse {
  chunk_a: { chunk_id: string; char_count: number; element_count: number }
  chunk_b: { chunk_id: string; char_count: number; element_count: number }
  deleted_chunk_id: string
}

export interface LinkResponse {
  group_id: string
  chunk_ids: string[]
}

export interface UnlinkResponse {
  unlinked_count: number
}

export const listChunks = (docId: string, params: { page?: number; size?: number }) =>
  request.get<any, ChunkListResponse>(`/documents/${docId}/chunks`, { params })

export const getChunkDetail = (chunkId: string) =>
  request.get<any, ChunkDetail>(`/chunks/${chunkId}`)

export const editChunk = (chunkId: string, fullText: string) =>
  request.put<any, EditChunkResponse>(`/chunks/${chunkId}`, { full_text: fullText })

export const deleteChunk = (chunkId: string) =>
  request.delete<any, { message: string; chunk_id: string }>(`/chunks/${chunkId}`)

export const mergeChunks = (chunkIds: string[]) =>
  request.post<any, MergeResponse>('/chunks/merge', { chunk_ids: chunkIds })

export const splitChunk = (chunkId: string, splitAt: number, linkGroup = false) =>
  request.post<any, SplitResponse>(`/chunks/${chunkId}/split`, {
    split_at: splitAt,
    link_group: linkGroup,
  })

export const linkChunks = (chunkIds: string[]) =>
  request.post<any, LinkResponse>('/chunks/link', { chunk_ids: chunkIds })

export const unlinkChunks = (chunkIds: string[]) =>
  request.post<any, UnlinkResponse>('/chunks/unlink', { chunk_ids: chunkIds })
